templates/cluster-templates/bases/hcloud-mt-md-0-ubuntu-ipv6-only.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: HCloudMachineTemplate
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
```

Comment:
I think we should give the machine deployment a different name, so that the name does not clash with the other MD. I suggest "${CLUSTER_NAME}-md-ipv6only-hcloud".